docs/source/ddp_comm_hooks.rst

```diff
@@ -185,11 +187,12 @@ Here is a simple, end-to-end example of saving and reloading PowerSGD state and
         map_location = {'cuda:%d' % 0: 'cuda:%d' % rank}
         checkpoint = torch.load(CHECKPOINT, map_location=map_location)
 
-        ddp_model.load_state_dict(checkpoint['state_dict'])
+        new_ddp_model = DistributedDataParallel(SimpleModel().to(rank), device_ids=[rank])
```

Comment:
why does there need to be a new model instance created?

Reply:
Since register_comm_hook or register_builtin_comm_hook can only be called once, a new ddp model needs to be created for using `register_comm_hook`

https://github.com/pytorch/pytorch/blob/79e0a1eacba67be881456b3e9d260634329f3836/torch/csrc/distributed/c10d/reducer.cpp#L1775-L1780

Reply:
Thanks for explaining!